File: JavaScript Fundamentals/Mid Exams/Mid Exam 5/inventory.js
function solve(arr) {
    let items = arr.shift().split(', ');
    for (let el of arr) {
        if (el == 'Craft!') break;

        let tokens = el.split(' - ');
        if (tokens[0] == 'Collect') {
            if (!(items.includes(tokens[1]))) items.push(tokens[1]);
        }
        else if (tokens[0] == 'Drop') {
             if (items.includes(tokens[1])) items.splice(items.indexOf(tokens[1]), 1);
        }
        else if (tokens[0] == 'Combine Items') {
            let givenItems = tokens[1].split(':');
            if (items.includes(givenItems[0])) items.splice(items.indexOf(givenItems[0]) + 1, 0, givenItems[1]);
        } else {
            if (items.includes(tokens[1])) {
                let removedItem = items.splice(items.indexOf(tokens[1]), 1);
                items.push(removedItem);
            }
        }
    }
    console.log(items.join(', '));
}

File: Python Fundamentals/Mid Exam problems/inventory.py
journal = input().split(', ')

while True:
    command = input()
    if command == "Craft!":
        print(', '.join(journal))
        break
    command_list = command.split(' - ')
    if command_list[0] == 'Collect':
        if command_list[1] not in journal:
            journal.append(command_list[1])
    elif command_list[0] == 'Drop':
        if command_list[1] in journal:
            journal.remove(command_list[1])
    elif command_list[0] == 'Combine Items':
        if command_list[1].split(':')[0] in journal:
            journal.insert(journal.index(command_list[1].split(':')[0]) + 1, command_list[1].split(':')[1])
    else:
        if command_list[1] in journal:
            journal.append(journal.pop(journal.index(command_list[1])))
